Bio/PDB/Mutate.py

```diff
@@ -0,0 +1,561 @@
+            lowest_energy = vdw_energy
+            best_rotamer = rotamer
+    return best_rotamer
+
+
+def mutate(
+    pdb_obj,
+    chain,
+    res_num,
+    mutate_to,
+    rotamer_lib=None,
+    mutation_type="best",
+):
+    """
+    Mutates a given residue based on the best fitting rotamers
+    """
+    Polypeptide.Polypeptide(
+        pdb_obj[0][chain]
+    ).get_phi_psi_list()  # This generates the xtra attribute for PHI and PSI angles for residues
+    try:
+        _residue = pdb_obj[0][chain][res_num]
+    except KeyError:
+        raise KeyError(f"Residue {res_num} not found in chain {chain}!")
+    for atom in list(_residue.get_atoms()):  # Create a copy to remove from
+        if not is_backbone(atom):
+            atom.parent.detach_child(atom.id)
+    phi, psi = (
+        round(np.rad2deg(y), -1) if y else 0
+        for y in [_residue.xtra[x] for x in ["PHI", "PSI"]]
+    )
+    # GET_ATR IS WRONG
+    sample_residue = read_sample_residue(mutate_to)
+    starting_points = np.asmatrix(
+        [sample_residue["N"], sample_residue["CA"], sample_residue["C"]]
+    )
+    end_points = np.asmatrix(
+        [_residue["N"].coord, _residue["CA"].coord, _residue["C"].coord]
+    )
+
+    sup = SVDSuperimposer.SVDSuperimposer()
+    sup.set(end_points, starting_points)
+    sup.run()
+    rot, tran = sup.get_rotran()
+
+    for atom, coords in sample_residue.items():
+        sample_residue[atom] = np.squeeze(np.asarray(np.dot(coords, rot) + tran))
+
+    if mutate_to not in ["ALA", "GLY"]:
+        if not rotamer_lib:
+            rotamer_lib = load_rotamers()
+        if mutation_type == "first":
+            selected_rotamer = sorted(
+                rotamer_lib[mutate_to][phi][psi], key=lambda x: x["prob"], reverse=True
+            )[0]
+        elif mutation_type == "random":
+            p = np.array([x["prob"] for x in rotamer_lib[mutate_to][phi][psi]])
+            p /= p.sum()
+            selected_rotamer = np.random.choice(rotamer_lib[mutate_to][phi][psi], p=p)
+        elif mutation_type == "best":
+            selected_rotamer = select_best_rotamer_based_on_clashes(
+                pdb_obj,
+                chain,
+                res_num,
+                mutate_to,
+                sample_residue,
+                rotamer_lib[mutate_to][phi][psi],
+            )
+        elif mutation_type == "bestother":
+            selected_rotamer = select_best_rotamer_based_on_clashes(
+                pdb_obj,
+                chain,
+                res_num,
+                mutate_to,
+                sample_residue,
+                rotamer_lib[mutate_to][phi][psi],
+                skip_own_chain=True,
+            )
+        else:
+            raise ValueError(
+                f"Unknown mutation type {mutation_type}. Possible choices are 'first', 'random', 'best', 'bestother'"
+            )
+
+        # Introduce the rotamer
+        for angle in ["CHI1", "CHI2", "CHI3", "CHI4"]:
```

Comment:
CHI angles are already defined in `ic_data.py` (along with some other useful stuff).

Reply:
I was able to find the reference planes in the `ic_data.py` file, however I cannot find the given axis for rotation. Should I add it into that file, or am I just blind?